source/beamline/BioXAS/BioXASCarbonFilterFarm.cpp

```diff
@@ -1,32 +1,15 @@
 #include "BioXASCarbonFilterFarm.h"
-#include "util/AMErrorMonitor.h"
 
 BioXASCarbonFilterFarm::BioXASCarbonFilterFarm(const QString &name, QObject *parent) :
-	BioXASBeamlineComponent(name, parent)
+	BioXASCarbonFilterFarmControl(name, parent)
 {
-	// Initialize local variables.
+	setUpstreamActuator(new BioXASCarbonFilterFarmActuatorControl(objectName()+"UpstreamActuator", this));
```

Comment:
Yes, I think that's how AMControl handles setting names that are given. I chose to use objectName() here instead of name() just to get around a variable naming conflict, as I think both functions return the same thing. Ha. A better/more clear idea would have been to rename the variable. Will make that change.